### PR TITLE
Ammend camera and add back v2_old

### PR DESCRIPTION
**Pull request by lpiccinelli-eth**

### CODE REVIEW

**Comment by GaoQi-2023** [comment]
After two days of use, I can confidently say that UniDepth v2 is indeed excellent. However, I've encountered a few unexpected issues during setup and usage that I'd like to report:

### 1. GCC Version Issue for PyTorch3D Installation
PyTorch3D requires GCC 11 or earlier, but Ubuntu 24.04 comes with GCC 13 by default, making installation impossible without adjustments. Users need to run the following commands (in an Anaconda environment):

```bash
conda install -c conda-forge gcc_linux-64=11 gxx_linux-64=11
ln -sf $CONDA_PREFIX/bin/x86_64-conda-linux-gnu-gcc $CONDA_PREFIX/bin/gcc
ln -sf $CONDA_PREFIX/bin/x86_64-conda-linux-gnu-g++ $CONDA_PREFIX/bin/g++
```

### 2. CUDA Compatibility for xFormers
Running `pip install xformers==0.0.24` directly installs a version compatible with CUDA 12.1 by default. Since CUDA 11.8 is recommended by the authors, users should instead install xFormers using:

```bash
pip install xformers==0.0.24 --index-url https://download.pytorch.org/whl/cu118
```

### 3. Errors When Exporting ONNX Models
Exporting UniDepth v2 to ONNX produces errors due to inconsistencies in the JSON filenames under the `configs/` directory and incorrect export commands in `assets/docs/V2_README.md`. The filenames should be corrected to:
- `config_v2_vits14.json`
- `config_v2_vitb14.json`
- `config_v2_vitl14.json`

Then the export command should be:

```bash
python ./unidepth/models/unidepthv2/export.py --version v2 --backbone vitl14 --shape 462 616 --output-path unidepthv2.onnx
```

### 4. Specifying Camera Intrinsic Matrix in ONNX Inference
Due to my limited experience, even with ChatGPT's assistance, I'm currently unable to add the option to specify a camera intrinsic matrix as input during UniDepth v2 inference, especially in the ONNX model, as this involves modifying the decoder logic. I hope the authors can address this functionality in future updates.

Thanks again for your excellent work!

**Comment by GaoQi-2023** [comment]
Additionally, I'd like to ask whether specifying the ground-truth camera matrix during video inference could help reduce the "flickering" between adjacent frames, which causes noticeable jitter of the point cloud along the Z-axis. This issue is particularly critical for my use case since I'm calculating object volumes using a projection method, where the meshes are projected onto a fixed reference plane at z = z0. Significant jitter along the Z-axis introduces substantial noise to the measurements.

I've previously experimented with Metric3D v2, but the extent of jitter output by the model forced me to abandon all mitigation strategies.

I would greatly appreciate your insights on this matter. Thanks again!

**Comment by lpiccinelli-eth** [comment]
First thank you for your appreciation! 

1. Pytorch3D: I will drop the dependency and "force" to install the custom KNN via `cd unidepth/ops/knn with && bash compile.sh`
2. CUDA for xFormers: we also worked with CUDA 12.1 and everything is fine, I will amend the readme with CUDA 11.8+
3. ONNX export: thank you for pointing this out, I will amend it in the next PR and also add the possibility to pass intrinsics to the ONNX export (I will need to define another class)
4. Flickering: I would suggest to compute the median camera over all your images if they are taken from the same camera, that usually helps stabilization, but some flickering is inherent in the model itself. I am planning to release a "stabilized" version (3D conv / attention + trained on videos).
